samples/bluetooth/bap_broadcast_sink/src/audio_codec.c

```diff
@@ -0,0 +1,172 @@
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <errno.h>
+#include <stdbool.h>
+#include <stddef.h>
+#include <stdint.h>
+#include <string.h>
+
+#include <zephyr/autoconf.h>
+#include <zephyr/device.h>
+#include <zephyr/devicetree.h>
+#include <zephyr/kernel.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/audio/codec.h>
+
+LOG_MODULE_REGISTER(codec, CONFIG_LOG_DEFAULT_LEVEL);
+
+/*
+ * Size of one audio block transferred to the codec, in bytes.
+ *
+ * For the configuration used here (16bit samples, mono channel),
+ * 480 bytes correspond to 240 PCM samples per block. This value was
+ * chosen as a compromise between:
+ *  - keeping latency reasonably low (smaller blocks)
+ *  - avoiding excessive interrupt / DMA overhead (larger blocks).
+ * If the PCM format (sample width or number of channels) is changed,
+ * this constant should be revisited so that the block size remains
+ * appropriate for the target use case and hardware.
+ */
+#define CODEC_BLOCK_SIZE 480
+
+/*
+ * Default speaker/output volume sent via AUDIO_PROPERTY_OUTPUT_VOLUME.
+ *
+ * The valid numeric range and exact mapping to dB are codec‑dependent;
+ * refer to the specific codec driver / datasheet for details. The value
+ * 15 was selected as a conservative, comfortable default for this sample
+ * application. Adjust as needed to match the desired default output level.
+ */
+#define SPEAKER_VOL      15
+
+/*
+ * Total size of the ring buffer used to queue audio data for the codec.
+ *
+ * The buffer is sized to hold 20 CODEC_BLOCK_SIZE blocks. Having multiple
+ * blocks queued reduces the likelihood of underruns when the producer is
+ * briefly delayed, while still keeping overall latency and RAM usage
+ * bounded. Increasing the multiplier adds buffering (and latency / RAM
+ * consumption); decreasing it reduces buffering but makes underruns more
+ * likely on a busy system.
+ */
+#define RING_BUF_SIZE    (CODEC_BLOCK_SIZE * 20)
+
+#if DT_NODE_HAS_STATUS_OKAY(DT_ALIAS(codec0))
+static uint8_t ring_buffer[RING_BUF_SIZE];
+static struct ring_buf rb;
+static bool codec_configed;
+static uint8_t block_data[CODEC_BLOCK_SIZE];
+static const struct device *g_dev;
+
+static void tx_done(const struct device *dev, void *user_data)
+{
+	uint32_t avail;
+	uint32_t read;
+
+	avail = ring_buf_size_get(&rb);
+	if (avail < CODEC_BLOCK_SIZE) {
+		LOG_WRN("Buffer empty\n");
+		return;
+	}
+	read = ring_buf_get(&rb, block_data, CODEC_BLOCK_SIZE);
+	if (read != CODEC_BLOCK_SIZE) {
+		LOG_WRN("Failed to read full block from ring buffer: %u bytes\n", read);
+		return;
+	}
+	audio_codec_write(dev, block_data, CODEC_BLOCK_SIZE);
+}
+int audio_codec_open(void)
+{
+	ring_buf_init(&rb, RING_BUF_SIZE, ring_buffer);
+	g_dev = DEVICE_DT_GET(DT_ALIAS(codec0));
+	if (!device_is_ready(g_dev)) {
+		LOG_ERR("Could not open codec device\n");
+		return -EIO;
+	}
+	return 0;
+}
+int audio_codec_cfg(uint32_t samplerate)
+{
+	int ret;
+	audio_property_value_t val;
```

Comment:
```suggestion
	const audio_property_value_t val = {.vol = SPEAKER_VOL};
```

Consider making this constant

Reply:
Updated, but do not use const, as audio_codec_set_property take non-const parameter.

Reply:
> Updated, but do not use const, as audio_codec_set_property take non-const parameter.

Since it's pass by value, that shouldn't really matter. It's not any different than
```c
void foo(int a)
{
    LOG_DBG("%d", a);
}

void bar(void)
{
    const int b = 10;
    foo(b);
}
```

Do you get a compiler error/warning?